src/contracts/GPv2Settlement.sol

```diff
@@ -51,6 +52,43 @@ contract GPv2Settlement {
         uniswapFactory = uniswapFactory_;
     }
 
+    /// @dev Execute a Uniswap trade exchanging the given amount of tokenIn
+    /// for the given amount of tokenOut.
+    ///
+    /// @param tokenIn The token sold on Uniswap.
+    /// @param tokenOut The token bought on Uniswap.
+    /// @param amountIn The amount sold.
+    /// @param amountOut The amount bought.
+    function uniswapTrade(
+        IERC20 tokenIn,
+        IERC20 tokenOut,
+        uint256 amountIn,
+        uint256 amountOut
```

Comment:
Instead of passing in both tokens and amounts, should we just pass in the `IUniswapV2Pair` and signed amounts? The reason I suggest this is:
1. The `IUniswapV2Pair` already needs to be computed in order to retrieve and increment the `nonce`
2. The amounts are already passed in as signed integers (negative is a buy amount, positive is a sell amount).

Alternatively, we could modify the `settle` method to use `tokenSurplus` `tokenDeficit` and then use the ordering of those tokens to determine `token0` and `token1` in a similar fashion to what is done here. Either way, I would rather both methods encode the Uniswap trade direction in the same way.

Reply:
> With this function I was hoping to abstract away the need for the caller to understand if tokenIn and tokenOut are Uniswap's token 0 or token 1. This is especially nice if we are going to include paths through ETH.

This is a very good argument for encoding swap direction in the `token*` inputs.

> Then the whole notion of token0 and token1 could be removed everywhere else by having the function uniswapPairAddress sort the addresses by itself, instead of reverting if the order is not correct.

This was more the point I was trying to make. Either we should encode the swap direction in the token address ordering (as you did here) or using the sign of the reserve balancing amounts (as I had originally done) and not mix and match. Given your previous argument, I also prefer it done the way it is here.